analyzers/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -83,7 +106,62 @@ public void ByDefault_ExecuteRule()
                 Verifier.VerifyAnalyzer(testCase.Path,
                                         testCase.Analyzer,
                                         ParseOptionsHelper.FromCSharp8,
-                                        MetadataReferenceFacade.SystemComponentModelPrimitives);
+                                        ConcatReferences(testCase.AdditionalReferences));
+            }
+        }
+
+        [TestMethod]
+        public void WhenProjectType_IsTest_RunsOnlyTestRules()
+        {
+            var sonarProjectConfig = TestHelper.CreateSonarProjectConfig(Directory.CreateDirectory(@"TestCases\WhenProjectType_IsTest_RunsOnlyTestRules").FullName, ProjectType.Test);
+            foreach (var testCase in testCases)
+            {
+                var isTest = testCase.Analyzer.SupportedDiagnostics.Any(d => d.CustomTags.Contains(DiagnosticDescriptorBuilder.TestSourceScopeTag));
+                if (isTest)
+                {
+                    // the rules with TEST scope should run
+                    Verifier.VerifyAnalyzer(testCase.Path,
+                                            testCase.Analyzer,
+                                            ParseOptionsHelper.FromCSharp8,
+                                            ConcatReferences(testCase.AdditionalReferences),
+                                            sonarProjectConfig);
+                }
+                else
+                {
+                    // MAIN-only rules
```

Comment:
_Nitpick:_ Just to align with the comment from the test below
```suggestion
                    // MAIN-only rule
```